### PR TITLE
Featuremilestone3 branch

### PR DESCRIPTION
**Pull request by em-em-D**
Milestone3 requirement have been implemented fully
As in my previous code review, Linter is still not working on my repository, having a slight linting error

### CODE REVIEW

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Changes Required❌.
# 🙋‍♂️ Michael
#### You have done a good job on this one, here are a few issues we have to address before the project could be approved🤗 .
* [ ] Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))
- You have not followed all the steps for setting up your linters as stated in the requirement, kindly visit this [page](https://github.com/microverseinc/linters-config/tree/master/react-redux), take note of step one. Let us address these issues once and for all.
### That is it kindly address the issue and submit a re-review👨‍🔧.
#### Happy coding👨‍💻
Addo Forison

**Comment by em-em-D** [comment]
Hi have followed all the instructions on the linter repository but still to no avail, linter is still not working, I have tried to fix it for the past 3 days now and hasn't worked, I have moved on to milestone 4

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi @em-em-D :hand: 
Great job on this milestone so far :clap: . We only need to make a small change before moving on. 

- [ ] We need to improve your README file with relevant information about your project.
Screenshot from 2020-09-08 11-59-54

- [ ] Microverse has switched from the stickler (stickler.ci) to Github actions as it described [here](https://microverse.zendesk.com/hc/en-us/articles/360049654493-June-4th-2020-GitHub-Actions-is-replacing-Stickler). We need to apply that change to your project :smile: . The Github Actions instructions can be found [here](https://github.com/microverseinc/linters-config/tree/master/react-redux)
You can safely delete the `.stickler` files

- [ ] After switch the linter solution to Github Actions we need to make sure that all tests pass.

Please kindly submit for code review again after making these changes.
Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @em-em-D  :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

**Kindly check out my comments below;**

**let's keep it simple, kindly work on the following;**
**The previous reviewer already mentioned this;**

### 1. 
**Let's make your repo looks professional by applying all the 3 basic rules, kindly update your `README` file and remove unnecessary content**.
Checkout out the below link for help, free free to follow this [template](https://github.com/microverseinc/readme-template) from pathwright

image

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/

### 2.
 **NOTE:** Microverse has moved on to Github actions for continuous deployment please use that instead of a stickler. 
image

https://github.com/microverseinc/linters-config/tree/master/javascript


**I'm pretty sure after these 2 things, you will get your PR approval! 😄** 


### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by em-em-D** [comment]
hello, sean bellom,
I Have set up what is required except that linters are not working on my project for some time now,
and secondly what do you mean by dedicated repository and professional, is my repo not professional? if you think so then contact microverse to change their guidelines cause it the guidelines they have provided that have been followed,
please I have done my best for 2 weeks now

**Comment by bellom** [comment]
Hi @em-em-D, 

Apparently you have tried to fix the linter's problem, understandable.
Let's keep it simple;
1. You should not push the `node_modules` file to the repo.
2. Let's set up the Github Action 👉  https://github.com/microverseinc/linters-config/tree/master/react-redux


Don't get me wrong, your repo looks good, but we need to make it look better by updating your `README`, seems you didn't check the README template I shared above, if you do you will know that your README file looks good but there is still a lot of room for improvement.

**Comment by em-em-D** [comment]
I have tried and tried and tried to fix eslint but to no avail if you can fix it kindly do, I have moved on to the next milestone

**Comment by bellom** [comment]
> I have tried and tried and tried to fix eslint but to no avail if you can fix it kindly do, I have moved on to the next milestone

Yeah, it's obvious you have been trying hard on it, it is understandable to move to the next milestone but before that, we could sync on a call to look into it together because I think the rest of the milestone might be having linter's issue too.

By the way, 
1. have you try to run `npx eslint .` 
2. you're missing `stylelint` config file in your repo.
3. try to `fix indentation and spacing`

File: src/App.js
Comment (uryelah) [inline src/App.js:1]: - [ ] Please remove unused files, you are using the components/App.js file, not this one in index.js.
Please remove it so you don't have two App.js files in your repo.

File: src/components/App.js
Comment (uryelah) [inline src/components/App.js:13]: - [ ] [OPTIONAL] Give the class name 'button' a more descriptive name like button-container or even just buttons.

File: src/components/Button.js
Comment (uryelah) [inline src/components/Button.js:7]: - [ ] Independent of the color the button should also have a border property in CSS.

File: src/components/ButtonPanel.js
Comment (uryelah) [inline src/components/ButtonPanel.js:12]: - [ ] [OPTIONAL] Think on a way to repeat less code, the <Button> element em specific, you can try adding a loop to make the code drier.

File: README.md
Comment (uryelah) [inline README.md:1]: - [ ] [OPTIONAL] I'd strongly encourage you to update your README file to the template linked by my colegues, the reason for that is that you'll most likelly have to do that in the future in career/interview prep and that a better README file can make recruiters more prone to contact you about jobs and opportunities.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Michael Damulak**, I'm Sarah your reviewer for this PR 👩‍🚀 

I know you must be tired of the linting checks by now but let's revisit them again because you have actually made progress since the initial issue.

The linters are set and the GitHub actions are working, the thing now is that your code is not passing all of the code standards and style checks from eslint.

If you click in the failing checks for details you'll be able to see the problematic lines and which rules they are breaking as seen in the screenshots below:

Screenshot_2020-09-15 Featuremilestone3 branch by em-em-D · Pull Request #3 · em-em-D React-calculator
Screenshot_2020-09-15 Featuremilestone3 branch by em-em-D · Pull Request #3 · em-em-D React-calculator(1)

You'll be able to see the same issue sin your terminal at the root of your project directory.

Just use the CLI commands found at: [https://eslint.org/docs/user-guide/command-line-interface](https://eslint.org/docs/user-guide/command-line-interface);

For the errors at the ServiceWorker, this file is not really needed for this project, so you can just delete it, while making sure it's not being imported anywhere. 
The other issues you can check one by one and fix them, some of them you can even fix automatically if you use the eslint command for that.

Again, check the [eslint command's documentation ](https://eslint.org/docs/user-guide/command-line-interface)and fix the linting issues. Those should be noted in all your milestones so the earlier you fix them the better. Also, make sure you always check for linting issues before pushing your code and opening a PR.

Rock on! 🎸 👨‍🎤 

# Required Changes 🏗️ 🐛 💡

- [ ] Make sure the linter checks don't output any error or warning.

- [ ] Add the .stylelintrc.json, [from the linters repo](https://github.com/microverseinc/linters-config/tree/master/react-redux), to your project root.

(Check the comments below for the other fixes)

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by em-em-D** [comment]
Hi Sarah,
Thanks for the code review but there is nothing you are saying that hasn’t
been done or tried. The service file is needed because you can’t run “npm
start” without it, it would trow and error, and when eslint is installed
locally to adjust styling errors  npm start doesn’t work.
All have been tried . I suggest you try it on your local machine as and see
if it works. Cheers

**Comment by uryelah** [comment]
@em-em-D I tested it in my machine and it seems to be possible to do the pointed fixes. 
Maybe there's a configuration issue going on in your local one.
You can schedule a call with a Fulltime TSE to get more help in solving this issue https://microverse.zendesk.com/hc/en-us/articles/360026625973-How-can-I-get-help-from-TSEs-.

EDIT: Please attempt some changes before asking for a new code review.
You can also ping me tomorrow, or any other TSE, on Slack and ask for their help/input.

File: src/components/App.js
Comment (uryelah) [inline src/components/App.js:14]: - [ ] Where is this.handleClick coming from? 
Make sure all the functions you are used are declared somewhere.

- [ ] App should be a functional component:
appFunctionalComponent

**Review by uryelah — CHANGES_REQUESTED** [review]
Please reach out to a TSE, or even a teammate, to help you debug the code. 
It's always possible to fix bugs and issues!

**Comment by em-em-D** [comment]
I am delighted to say that everything works perfectly well and all linters and style errors have passed

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED

Hi @em-em-D good job man, you have done really well to follow up with this and make sure it works properly, whoops that wasn't easy :slightly_smiling_face:  please merge and move on **This is approved** :slightly_smiling_face: 

**RAPHAEL NORIODE** :mask: STAY_SAFE

**Comment by em-em-D** [comment]
milestone 3 done

**Comment by em-em-D** [comment]
milestone 3 done and dusted

**Pull request closed without merging**
